src/commonMain/kotlin/fr/acinq/lightning/payment/IncomingPaymentHandler.kt

```diff
@@ -402,15 +402,11 @@ class IncomingPaymentHandler(val nodeParams: NodeParams, val db: IncomingPayment
     }
 
     /**
-     * If we are disconnected, the LSP will forget pending pay-to-open requests. We need to do the same otherwise we
-     * will accept outdated ones.
+     * If we are disconnected, we must forget pending payment parts.
+     * Pay-to-open requests will be forgotten by the LSP, so we need to do the same otherwise we will accept outdated ones.
+     * Offered HTLCs that haven't been resolved will be re-processed when we reconnect.
      */
-    fun purgePayToOpenRequests() {
-        val valuesToReplace = pending.mapValues { entry -> entry.value.copy(parts = entry.value.parts.filter { it !is PayToOpenPart }.toSet()) }
-        pending.plusAssign(valuesToReplace)
-        val keysToRemove = pending.filterValues { it.parts.isEmpty() }.keys
-        pending.minusAssign(keysToRemove)
-    }
+    fun purgePendingPayments() = pending.clear()
```

Comment:
That was easy! 🧹 

Some logging would be nice, also in `purgeExpiredPayments`, to have a better idea of what is going on when debugging.

Reply:
Good idea, done in https://github.com/ACINQ/lightning-kmp/pull/583/commits/f3b79f5a4f17a60bfb10a73674314bb37b7e826d